src/components/map/ha-map.ts

```diff
@@ -155,8 +155,8 @@ export class HaMap extends ReactiveElement {
   }
 
   private _updateMapStyle(): void {
-    const darkMode = this.darkMode ?? this.hass.themes.darkMode ?? false;
-    const forcedDark = this.darkMode ?? false;
+    const darkMode = this.hass.themes.darkMode ?? false;
+    const forcedDark = this.darkMode ?? this.hass.themes.darkMode ?? false;
```

Comment:
```suggestion
    const forcedDark = this.darkMode ?? false;
```
I would suggest to not use the theme mode for forced dark because the "forced-dark" class is only use for map color filter. The map filter for dark theme is already handled by theme. We should not override this in the map component because it will break the ability to the user to set a custom map filter using theme.
https://github.com/home-assistant/frontend/blob/7167b66719a4b496a3e6596d8a6d1a35aedf7624/src/resources/styles-data.ts#L53-L54

May be we should rename the variable to "darkTheme` and `darkMapForced` to make it cleared? WDYT?

Reply:
I missed the fact that `map-filter` variable may be set in a custom theme.
So, I agree to your correction, thank you.
Also I agree to rename `darkTheme` & `darkMapForced` as you suggested.
Also - suggest to rename `className` to `classNameMarker`.

What I do not like: assume a user has a dark theme with a `--map-filter` which differs from our `invert(.9) hue-rotate(170deg) brightness(1.5) contrast(1.2) saturate(.3)`.
Means - in the "forced dark" mode the filter may be different than defined by a user in a dark theme.
So, I am thinking about adding `--map-filter-light` & `--map-filter-dark`.

Reply:
From another side, your proposal to set `forcedDark` independently on a theme is valid if we do no need to set a light map in a dark theme. My initial goal was to allow this scenario... I will think again how to achieve my proposed scenario with a fact that the `--map-filter` variable may be set in a theme.

Reply:
The property is called `dark_mode` (see [map card documentation](https://www.home-assistant.io/dashboards/map/)). It should not force the map to light if it's set to `false`. Also, there is no way to set it to `undefined` using the UI. 
If we want to do it properly, we should have a `map_theme` config option with 3 possibles options (auto, light, dark) but I'm not sure it's really needed.

For the different map filters I agree, we can try to use the `map-filter` variable defined in the current theme if there is a dark themes defined, otherwise we default to the default map filter.

Reply:
> It should not force the map to light if it's set to false

Shouldn't it be treated as "force_dark_mode" then? Ofc, I mean only a description of this option, we should not rename the option to `force_dark_mode` for compatibility. Now the option is described as "Enable a dark theme for the map" - probably it should be changed to "Enable a dark mode for the map in a light theme".

> we can try to use the map-filter variable defined in the current theme if there is a dark themes defined, otherwise we default to the default map filter

I am still thinking about it. Someone may want a custom theme with the `--map-filter` defined and this theme may be not defined as "dark".

Reply:
The rules must be :
If the boolean is true, apply dark theme.
If the boolean is false, do nothing.

There is no need to change the name or the property.

Reply:
Please check my revised PR.